cmd/generic-handlers.go

```diff
@@ -183,19 +186,17 @@ func shouldProxy() bool {
 }
 
 func (h redirectHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if guessIsRPCReq(r) || guessIsBrowserReq(r) ||
+	if !shouldProxy() || guessIsRPCReq(r) || guessIsBrowserReq(r) ||
 		guessIsHealthCheckReq(r) || guessIsMetricsReq(r) || isAdminReq(r) {
 		h.handler.ServeHTTP(w, r)
 		return
 	}
-	if shouldProxy() {
-		// if this server is still initializing, proxy the request
-		// to any other online servers to avoid 503 for any incoming
-		// API calls.
-		if idx := getOnlineProxyEndpointIdx(); idx >= 0 {
-			proxyRequest(context.TODO(), w, r, globalProxyEndpoints[idx])
-			return
-		}
+	// if this server is still initializing, proxy the request
+	// to any other online servers to avoid 503 for any incoming
+	// API calls.
+	if idx := getOnlineProxyEndpointIdx(); idx >= 0 {
+		proxyRequest(context.TODO(), w, r, globalProxyEndpoints[idx])
```

Comment:
```suggestion
		proxyRequest(r.Context(), w, r, globalProxyEndpoints[idx])
```
Not that the context is actually used in `proxyRequest`, but in the future we might.